diff --git a/lib/DAVACL/PrincipalBackend/BackendInterface.php b/lib/DAVACL/PrincipalBackend/BackendInterface.php
index 72717a59bf..7140a92954 100644
--- a/lib/DAVACL/PrincipalBackend/BackendInterface.php
+++ b/lib/DAVACL/PrincipalBackend/BackendInterface.php
@@ -110,7 +110,7 @@ public function searchPrincipals($prefixPath, array $searchProperties, $test = '
      * @param string $uri
      * @param string $principalPrefix
      *
-     * @return string
+     * @return string|null
      */
     public function findByUri($uri, $principalPrefix);
 
