diff --git a/src/CloudFront/Signer.php b/src/CloudFront/Signer.php
index f6a7ed6e36..dee68e024d 100644
--- a/src/CloudFront/Signer.php
+++ b/src/CloudFront/Signer.php
@@ -81,6 +81,7 @@ public function getSignature($resource = null, $expires = null, $policy = null)
         $signatureHash = [];
         if ($policy) {
             $policy = preg_replace('/\s/s', '', $policy);
+            self::validatePolicy($policy);
             $signatureHash['Policy'] = $this->encode($policy);
         } elseif ($resource && $expires) {
             $expires = (int) $expires; // Handle epoch passed as string
@@ -136,4 +137,35 @@ private function encode($policy)
     {
         return strtr(base64_encode($policy), '+=/', '-_~');
     }
+
+    /**
+     * Validates a customer provided json document.
+     *
+     * @param string $jsonPolicy
+     *
+     * @return void
+     */
+    private static function validatePolicy(string $jsonPolicy): void
+    {
+        $policy = json_decode($jsonPolicy, true);
+        foreach ($policy['Statement'] ?? [] as $statement) {
+            if (isset($statement['Resource'])) {
+                self::validateResourceUrl($statement['Resource']);
+            }
+        }
+    }
+
+    /**
+     * @param string $url
+     *
+     * @return void
+     */
+    private static function validateResourceUrl(string $url): void
+    {
+        if (preg_match('/["\\\\\x00-\x1F]/', $url)) {
+            throw new \InvalidArgumentException(
+                'URL contains invalid characters: ", \\, or control characters'
+            );
+        }
+    }
 }
